Assets/ModScripts/Submodules/PianoDecryption.cs

```diff
@@ -0,0 +1,226 @@
+        { "FAKERS", new int[] { 3, 3, 2, 2, 3, 3, 2, 3, 3, 2, 2, 3 } }, 
+        { "GARAGE", new int[] { 7, 7, 0, 7, 7, 0, 7, 0 } }, 
+        { "HORROR", new int[] { 1, 2, 4, 5, 1, 2, 4, 5, 10, 9 } }, 
+        { "JULIET", new int[] { 10, 9, 10, 5, 3, 10, 9, 10, 5, 3 } }, 
+        { "KEVLAR", new int[] { 7, 7, 7, 3, 10, 7, 3, 10, 7 } }, 
+        { "NETHER", new int[] { 1, 11, 9, 6, 8, 9, 8, 6 } }, 
+        { "OPIOID", new int[] { 7, 9, 7, 4, 7, 9, 7, 4 } }, 
+        { "PARENT", new int[] { 7, 7, 7, 7, 7, 7, 7, 10, 3, 5, 7 } }, 
+        { "QUESTS", new int[] { 6, 7, 9, 9, 2, 11, 9, 7, 4, 2 } }, 
+        { "UMPIRE", new int[] { 3, 3, 1, 8, 3, 3, 5, 1 } }, 
+        { "VICTOR", new int[] { 11, 9, 7, 3, 2, 9, 11, 9, 7 } }, 
+        { "WIRING", new int[] { 3, 5, 3, 0, 8, 5, 3 } }, 
+        { "XENONS", new int[] { 2, 2, 2, 1, 1, 1, 11, 1, 11, 6 } }, 
+        { "YIELDS", new int[] { 7, 4, 5, 7, 0, 11, 0, 2, 0, 11, 9, 7 } }, 
+        { "ZAMBIA", new int[] { 10, 9, 10, 7 } }
+    };
+    string[] WordList = { "ANSWER", "BANTER", "CREOLE", "DRIVEN", "ELEVEN", "FAKERS", "GARAGE", "HORROR", "JULIET", "KEVLAR", "NETHER", "OPIOID", "PARENT", "QUESTS", "UMPIRE", "VICTOR", "WIRING", "XENONS", "YIELDS", "ZAMBIA" };
+    string encryptedWord, decryptedWord = "";
+    int[] solutionSequence = new int[] { };
+    int repeats = 1;
+    int currentNote = 0;
+
+    public PianoDecryption(CruelModkitScript Module, int ModuleID, ComponentInfo Info, byte Components) : base(Module, ModuleID, Info, Components)
+    {
+        Debug.LogFormat("[The Cruel Modkit #{0}] Solving Piano Decryption.", ModuleID);
+        WordList = WordList.OrderBy(x => Random.Range(1, 1000)).ToArray();
+        EncryptWord();
+        solutionSequence = noteSequences[decryptedWord];
+        if (decryptedWord == "ZAMBIA") repeats = Random.Range(3, 7);
+        Debug.LogFormat("[The Cruel Modkit #{0}] Solution piano sequence: {1}{2}.", ModuleID, solutionSequence.Select(x => Info.PianoKeyNames[x]).Join(", "),
+            repeats > 1 ? (" (played " + repeats.ToString() + " times)") : "");
+    }
+
+    public override void OnPianoPress(int Piano)
+    {
+        if (Module.IsAnimating())
+            return;
+
+        Module.Audio.PlaySoundAtTransform(Module.PianoSounds[Piano + (Info.Piano * 12)].name, Module.transform);
+        Module.Piano[Piano].GetComponentInChildren<KMSelectable>().AddInteractionPunch(0.25f);
+
+        if (Module.IsModuleSolved())
+            return;
+
+        if (!Module.CheckValidComponents())
+        {
+            Debug.LogFormat("[The Cruel Modkit #{0}] Strike! The {1} key on the piano was pressed when the component selection was [{2}] instead of [{3}].", ModuleID, Info.PianoKeyNames[Piano], Module.GetOnComponents(), Module.GetTargetComponents());
+            Module.CauseStrike();
+            return;
+        }
+
+        Module.StartSolve();
+        if (Piano != solutionSequence[repeats > 1 ? currentNote % 4 : currentNote])
```

Comment:
```suggestion
        if (Piano != solutionSequence[repeats > 1 ? currentNote % solutionSequence.Count() : currentNote])
```
You've made it modular so that any word can have a reapeat sequence, which is good, however putting a constant here will make it work incorrectly for words other than ZAMBIA